extensions/walletProviders/kadena/src/extension.ts

```diff
@@ -0,0 +1,79 @@
+import { ITatumSdkContainer, Network, TatumConfig, TatumSdkWalletProvider } from '@tatumio/tatum'
+import { KadenaNetworkId, KadenaTxPayload, KadenaWallet } from './types'
+import { genKeyPair } from "@kadena/cryptography-utils/lib/genKeyPair";
+import { restoreKeyPairFromSecretKey } from "@kadena/cryptography-utils";
+import { ChainId, createClient, createSignWithKeypair, ICommand } from "@kadena/client";
+import { ICommandPayload } from "@kadena/types";
+import { KadenaLoadBalancerRpc } from "@tatumio/tatum/dist/src/service/rpc/other/KadenaLoadBalancerRpc";
+
+export class KadenaWalletProvider extends TatumSdkWalletProvider<KadenaWallet, KadenaTxPayload> {
+  private readonly kadenaRpc: KadenaLoadBalancerRpc
+  private readonly sdkConfig: TatumConfig
+
+  constructor(tatumSdkContainer: ITatumSdkContainer) {
+    super(tatumSdkContainer)
+    this.sdkConfig = this.tatumSdkContainer.getConfig()
+    this.kadenaRpc = this.tatumSdkContainer.get(KadenaLoadBalancerRpc)
+  }
+
+  /**
+   * Gets a Kadena wallet, which includes an address, secret key, and a public key.
+   * @returns {KadenaWallet} An object containing address, secret key, and public key.
+   */
+  public async getWallet() {
+    const keyPair = genKeyPair()
+    const address = 'k:' + keyPair.publicKey
+
+    return { address, publicKey: keyPair.publicKey, secretKey: keyPair.secretKey} as KadenaWallet
+  }
+
+  /**
+   * Signs and broadcasts a Kadena transaction payload.
+   * @param {KadenaTxPayload} txPayload - The Kadena transaction payload, which includes secret key and transaction details.
+   * @returns {Promise<string>} A promise that resolves to the transaction hash.
+   */
+  public async signAndBroadcast(txPayload: KadenaTxPayload): Promise<string> {
+    const keyPair = restoreKeyPairFromSecretKey(txPayload.secretKey)
+    const signWithKeypair = createSignWithKeypair(keyPair)
+    const payload: ICommandPayload = JSON.parse(txPayload.command.cmd)
+
+    const chainId = payload.meta.chainId
+
+    if(!chainId)
```

Comment:
Nitpick, but there seems to be some issue with the formatter. Please ensure prettier is run as there are many cases like this, `{ client}`, etc.

Reply:
Fixed